ADApp/pluginSrc/NDFileHDF5AttributeDataset.cpp

```diff
@@ -185,36 +185,61 @@ asynStatus NDFileHDF5AttributeDataset::writeAttributeDataset(hdf5::When_t whenTo
       extendIndexDataSet(offsets[indexed]);
     }
     // find the data based on datatype
-    ret = ndAttr->getValue(ndAttr->getDataType(), pDatavalue, MAX_ATTRIBUTE_STRING_SIZE);
+    ret = ndAttr->getValue(ndAttr->getDataType(), pDatavaluestore[dataset_count], MAX_ATTRIBUTE_STRING_SIZE);
     if (ret == ND_ERROR) {
-      memset(pDatavalue, 0, MAX_ATTRIBUTE_STRING_SIZE);
+      memset(pDatavaluestore[dataset_count], 0, MAX_ATTRIBUTE_STRING_SIZE);
+    }
+    dataset_count++;
```

Comment:
Is dataset_count the size of the current buffer of values to be written to the dataset? It could possibly do with a different name. It sounds like it is a number of (h5) datasets.

Reply:
I renamed it to attributeBatchCount_, do you think that's more appropriate?

Reply:
That's much clearer.